src/bls/bls.service.ts

```diff
@@ -0,0 +1,105 @@
+import * as blst from '@chainsafe/blst';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { Type, fromHexString } from '@chainsafe/ssz';
+import { ProviderService } from 'provider';
+import {
+  Inject,
+  Injectable,
+  OnModuleInit,
+  LoggerService,
+} from '@nestjs/common';
+import {
+  BLSPubkey,
+  Bytes32,
+  DOMAIN_DEPOSIT,
+  GENESIS_FORK_VERSION_BY_CHAIN_ID,
+  UintNum64,
+  ZERO_HASH,
+} from './bls.constants';
+import { DepositMessage, ForkData, SigningData } from './bls.containers';
+import { DepositData } from './interfaces';
+
+@Injectable()
+export class BlsService implements OnModuleInit {
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private logger: LoggerService,
+    private providerService: ProviderService,
+  ) {}
+
+  async onModuleInit() {
+    const chainId = await this.providerService.getChainId();
+    this.forkVersion = GENESIS_FORK_VERSION_BY_CHAIN_ID[chainId] ?? null;
+  }
+
+  private forkVersion: Uint8Array | null = null;
+
+  public verify(depositData: DepositData): boolean {
+    if (!this.forkVersion) {
+      throw new Error('Fork version is not set');
```

Comment:
# Common

It isn't a good idea to throw a common exception. 
[Motivation article. Point 2](https://dzone.com/articles/9-best-practices-to-handle-exceptions-in-java).
Another approve:
<img width="810" alt="Screenshot 2022-07-14 at 14 55 35" src="https://user-images.githubusercontent.com/102730938/178976807-f20ec36d-c0b5-4094-87e2-722909cbdfb9.png">

Code Complete: Exceptions

Reply:
I'm not sure it makes sense, as long as we don't catch this exception and check it to see if they belong to any class